freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/ums/BulkUmsUsersStateProvider.java

```diff
@@ -0,0 +1,137 @@
+package com.sequenceiq.freeipa.service.freeipa.user.ums;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.google.common.collect.Maps;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.freeipa.service.freeipa.user.conversion.FmsGroupConverter;
+import com.sequenceiq.freeipa.service.freeipa.user.conversion.FmsUserConverter;
+import com.sequenceiq.freeipa.service.freeipa.user.conversion.WorkloadCredentialConverter;
+import com.sequenceiq.freeipa.service.freeipa.user.model.EnvironmentAccessRights;
+import com.sequenceiq.freeipa.service.freeipa.user.model.FmsGroup;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UmsUsersState;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UsersState;
+import com.sequenceiq.freeipa.service.freeipa.user.model.WorkloadCredential;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider.INTERNAL_ACTOR_CRN;
+
+@Component
+public class BulkUmsUsersStateProvider extends BaseUmsUsersStateProvider {
+    @Inject
+    private GrpcUmsClient grpcUmsClient;
+
+    @Inject
+    private EnvironmentsRightsChecksFactory environmentsRightsChecksFactory;
+
+    @Inject
+    private FmsUserConverter fmsUserConverter;
+
+    @Inject
+    private WorkloadCredentialConverter workloadCredentialConverter;
+
+    @Inject
```

Comment:
I don't think you need inject on constructor

Reply:
ok. I'll try to inject directly into the base class.